### DISCUSSION TITLE
Release at version 1.4 or version 1.3.2

### DISCUSSION THREAD

**Discussion by jgadsden** [OWASP/threat-dragon#82] [General] [answered]
The last release was version 1.3.1 in October 2020 and we should prepare another release for early May 2021.

@lreading has made large changes to the organisation of the files etc, and there are next to no actual functional changes. 

@lreading , @andk123 , @mike-goodwin with no real functional change should the next release be version 1.3.2 or do we go for version 1.4?

**Comment by lreading** [comment, marked as answer]
I've always been a fan of semver.  If we have no real functional changes yet, my vote would be for 1.3.2 :smiley:

**Reply by jgadsden** [reply]
OK, sounds good - lets go with version 1.3.2

**Comment by jgadsden** [comment]
We are intending to release in the next few days, but this can be pushed back if we are in the middle of a development phase. I have  given [advance notice](https://owasp.org/www-project-threat-dragon/#div-releases) that v1.3.2 will be in May, but that is easily shuffled around if necessary

@lreading what would you prefer? I do not want to interrupt the work you are doing for a patch version release, on other other hand there have been many improvements to the source that would be good to tag with a release fairly soon.

**Reply by lreading** [reply]
Don't let me slow you down, do what you need to do!

I'd love to explore the possibility of a continuous delivery pipeline.  In a perfect world, most releases (at least for web, not sure about desktop) would be able to be automated.  Any thoughts on that? Is it worth a separate discussion?

**Reply by jgadsden** [reply]
It would be great to automate the release, and this is most applicable to the desktop version which runs on varied platforms and has signing issues etc. The web app version is basically a significant tag, whereas the desktop app has various installers, checksums and the like

I guess it depends on how confident we are on the test coverage of the desktop app, which could then give us the confidence for automation. At  the moment I go round and install desktop app on my (Trisquel) Linux box, MacOS and lastly a Windows 10 machine just to make sure signing etc is actually OK. This is a chore, absolutely

**Reply by jgadsden** [reply]
Having said that, @lreading makes a good point in #96 that if the docker file is published then this is more than just a tag  for the web app, and the docker image should really be built on the supported platforms and also the container tested. 

Worthwhile though, as the docker container will be used for CI/CD
